packages/devtools-components/src/tree.js

```diff
@@ -679,10 +812,37 @@ class Tree extends Component {
         return;
 
       case "Enter":
-        this._activateNode();
+      case " ":
+        if (this.treeRef.current === this.activeElement) {
+          this._preventEvent(e);
+          if (this.props.active !== this.props.focused) {
+            this._activate(this.props.focused);
+          }
+        }
+        return;
+
+      case "Escape":
+        this._preventEvent(e);
+        if (this.props.active != undefined) {
+          this._activate(undefined);
+        }
+
+        if (this.treeRef.current !== this.activeElement) {
+          this.treeRef.current.focus();
+        }
+        return;
     }
   }
 
+  /**
+   * Current active element within the document.
+   * @return {DOMNode}
+   *         Current active element within the document if available.
+   */
+  get activeElement() {
+    return this.treeRef.current.ownerDocument.activeElement;
```

Comment:
should we guard on `this.treeRef` ? (in case this is called before the component is mounted/after it's unmounted) ?

Reply:
I can add that!